src/libraries/System.Private.CoreLib/src/System/DateOnly.cs

```diff
@@ -0,0 +1,902 @@
+        public int CompareTo(DateOnly value)
+        {
+            if (_dayNumber < value._dayNumber) return -1;
+            if (_dayNumber > value._dayNumber) return 1;
+            return 0;
+        }
+
+        /// <summary>
+        /// Compares the value of this instance to a specified object that contains a specified DateOnly value, and returns an integer that indicates whether this instance is earlier than, the same as, or later than the specified DateOnly value.
+        /// </summary>
+        /// <param name="value">A boxed object to compare, or null.</param>
+        /// <returns>Less than zero if this instance is earlier than value. Greater than zero if this instance is later than value. Zero if this instance is the same as value.</returns>
+        public int CompareTo(object? value)
+        {
+            if (value == null) return 1;
+            if (value is not DateOnly dateOnly)
+            {
+                throw new ArgumentException(SR.Arg_MustBeDateOnly);
+            }
+
+            return CompareTo(dateOnly);
+        }
+
+        /// <summary>
+        /// Returns a value indicating whether the value of this instance is equal to the value of the specified DateOnly instance.
+        /// </summary>
+        /// <param name="value">The object to compare to this instance.</param>
+        /// <returns>true if the value parameter equals the value of this instance; otherwise, false.</returns>
+        public bool Equals(DateOnly value) => _dayNumber == value._dayNumber;
+
+        /// <summary>
+        /// Returns a value indicating whether this instance is equal to a specified object.
+        /// </summary>
+        /// <param name="value">The object to compare to this instance.</param>
+        /// <returns>true if value is an instance of DateOnly and equals the value of this instance; otherwise, false.</returns>
+        public override bool Equals(object? value)
+        {
+            if (value is DateOnly)
+            {
+                return _dayNumber == ((DateOnly)value)._dayNumber;
+            }
+            return false;
+        }
+
+        /// <summary>
+        /// Returns the hash code for this instance.
+        /// </summary>
+        /// <returns>A 32-bit signed integer hash code.</returns>
+        public override int GetHashCode() => _dayNumber;
+
+        // Only Allowed DateTimeStyles: AllowWhiteSpaces, AllowTrailingWhite, AllowLeadingWhite, and AllowInnerWhite
+
+        /// <summary>
+        /// Converts a memory span that contains string representation of a date to its DateOnly equivalent by using the conventions of the current culture.
+        /// </summary>
+        /// <param name="s">The memory span that contains the string to parse.</param>
+        /// <returns>An object that is equivalent to the date contained in s.</returns>
+        public static DateOnly Parse(ReadOnlySpan<char> s) => Parse(s, DateTimeFormatInfo.CurrentInfo, DateTimeStyles.None);
```

Comment:
It is better to use more informative parameter name since it is _public_.

Below too.

Reply:
We are keeping parity with the other Date and Time APIs.